### PR TITLE
OF-1232 Organization System

### PR DESCRIPTION
**Pull request by MAG-AdamThorn**
This PR introduces a new Organization System which will enable Organizations to own Spaces and Assets and to control access to them.
This PR introduces the ability to invite and bulk invite users to an Organization and to give them specific Organization Roles. In addition the api allows those with appropriate permissions to remove members from an Organization and to get the roles associated with a given member.
Tests have been added for the above.

**For the reviewer**

* [ ] If required, are the changes covered by appropriate tests?
* [ ] Are any public-facing API changes well documented?
* [ ] Is the code easily readable and extensible and/or follow existing conventions?

### CODE REVIEW

File: Library/include/CSP/Systems/Organizations/Organization.h
Comment (lyubomirlichev) [inline Library/include/CSP/Systems/Organizations/Organization.h:101]: Which OrganizationSystem API returns this result ? Based off this diagram here https://miro.com/app/board/uXjVNz7XMkQ=/?moveToWidget=3458764577696164303&cot=14
This would be needed as part of showing the user their organization name requirement.
Or potentially there are other means of getting the organization name?

Comment (MAG-AdamThorn) [reply Library/include/CSP/Systems/Organizations/Organization.h:101]: The diagram is out of date and will be updated. They will be able to call GetOrganization using the cached Id.

File: Library/src/Systems/Organizations/OrganizationSystem.cpp
Comment (MAG-AdamThorn) [reply Library/src/Systems/Organizations/OrganizationSystem.cpp:284]: It looked correct in VS but fixed any formatting issues with Ctrl+K, Dtrl+D

File: Library/src/Systems/Organizations/OrganizationSystem.cpp
Comment (MAG-AdamThorn) [reply Library/src/Systems/Organizations/OrganizationSystem.cpp:211]: It looked correct in VS but fixed any formatting issues with Ctrl+K, Dtrl+D

File: Library/src/Systems/Organizations/OrganizationSystem.cpp
Comment (MAG-AdamThorn) [reply Library/src/Systems/Organizations/OrganizationSystem.cpp]: It looked correct in VS but fixed any formatting issues with Ctrl+K, Dtrl+D

File: Library/src/Systems/Organizations/OrganizationSystem.cpp
Comment (MAG-AdamThorn) [reply Library/src/Systems/Organizations/OrganizationSystem.cpp]: It looked correct in VS but fixed any formatting issues with Ctrl+K, Ctrl+D

File: Library/src/Systems/Organizations/OrganizationSystem.cpp
Comment (MAG-AdamThorn) [reply Library/src/Systems/Organizations/OrganizationSystem.cpp]: It looked correct in VS but fixed any formatting issues with Ctrl+K, Ctrl+D

File: Library/src/Systems/Organizations/OrganizationSystem.cpp
Comment (MAG-AdamThorn) [reply Library/src/Systems/Organizations/OrganizationSystem.cpp]: It looked correct in VS but fixed any formatting issues with Ctrl+K, Ctrl+D

File: Library/src/Systems/Organizations/OrganizationSystem.cpp
Comment (MAG-AdamThorn) [reply Library/src/Systems/Organizations/OrganizationSystem.cpp:61]: It looked correct in VS but fixed any formatting issues with Ctrl+K, Ctrl+D

File: Library/include/CSP/Systems/Organizations/Organization.h
Comment (lyubomirlichev) [reply Library/include/CSP/Systems/Organizations/Organization.h:101]: Which API is GetOrganization as i am not spotting it in the system public endpoints ?

**Review by MAG-SamBirley — DISMISSED** [review]
LGTM but recommend getting the blessing of Mag client teams before merging.

File: Library/include/CSP/Systems/Organizations/Organization.h
Comment (MAG-AdamThorn) [reply Library/include/CSP/Systems/Organizations/Organization.h:101]: It hasn't been done yet. This PR relates to tickets OF-1232 and OF-1236 - see here: https://magnopus.atlassian.net/jira/software/c/projects/OF/boards/258?selectedIssue=OF-1207

Tickets still to be worked on are: OF-1233, OF-1237 and OF-1238.
The GetOrganization() endpoint will be part of OF-1237.

Comment (lyubomirlichev) [reply Library/include/CSP/Systems/Organizations/Organization.h:101]: Right, that makes sense, thank you for the clarification 🙂

**Review by lyubomirlichev — DISMISSED** [review]
LGTM

**Review by MAG-SamBirley — DISMISSED** [review]

**Comment by MAG-DanielMurdolo** [comment]
I think it would be nice to add the ability to pass in the Org ID you want to be affecting to most (if not all) of the methods. It gives the CSP user/developer more agency over knowing which org they are changing.

 Adam and I chatted through this on a call just now so we understood what each other was saying.

**Review by MAG-SamBirley — APPROVED** [review]
Nice work! Love the new `OrganizationID` param ❤

**Review by MAG-DanielMurdolo — APPROVED** [review]

**Pull request merged**
